ios/VisionCameraQrcodeScanner.swift

```diff
@@ -28,6 +27,21 @@ class VisionCameraQrcodeScanner: NSObject, FrameProcessorPluginBase {
         return barCodeAttributes
     }
     
+    static func createScanner(_ args: [Any]!) throws {
+        guard let rawFormats = args[0] as? [Int] else {
+            throw BarcodeError.noBarcodeFormatProvided
+        }
+        if (barcodeScanner == nil || barcodeRawFormats != rawFormats) {
```

Comment:
You probably don't want to be comparing array references directly (I imagine they'll never have reference equality). Since the barcode formats are all powers of 2, you could probably just bitwise `|` all the format options together to get a representative `int`, and compare that.

Reply:
Yeah you are right forgot about different ordering of options shouldn't create a new `BarcodeScanner` instance. Not really sure what you mean by "bitwise all the format options together" though? Do you want me to multiply them to get this representative int? Thanks!

Reply:
Because the values of all the different options are powers of 2, that means in a binary integer representation, each option corresponds to a single distinct bit position. This means that you can take an integer, loop through the options, and for each option "turn on" their respective bit in that integer with a bitwise OR operation. Something like:
```swift
var formatsBitmap: Int = 0
rawFormats.forEach { rawFormat in
    formatsBitmap = formatsBitmap | rawFormat
}
```
Then you can use `formatsBitmap` to detect a change in options based on a set-wise comparison (thanks to the bitmap), rather than a strict sequence comparison.

Reply:
@johankasperi  what dougkeen said make sense for you?, i'm not very familiar with swift but this make sense for me, lmk if you want to make any other changes, otherwise I'll do some tests and merge if everything works

Reply:
Sorry haven't really been able to look into this. Been busy with other stuff.

I'm afraid that I still don't really understand 😄 More exactly I don't understand how I should use the `formatsBitMap` after creating it. Would really appreciate if someone could provide some pseudo-code out of this for my understanding: "Then you can use formatsBitmap to detect a change in options based on a set-wise comparison (thanks to the bitmap), rather than a strict sequence comparison."

Another option could be to use `Set` in Swift: https://stackoverflow.com/a/29852086 If `set1.subtract(set2)` returns any objects then we would know that the formats arg has changed. But maybe that is not good performance wise?

Reply:
@johankasperi check your fork. I submitted a [PR](https://github.com/johankasperi/vision-camera-qrcode-scanner/pull/1) over there that implements what I’m talking about 😄

Reply:
@dougkeen  after i merge this PR you have time to send this update on a new PR?

Reply:
Yeah, we could do it that way 👍🏻

Reply:
#20